### PR TITLE
Fix cmd_func_set_pos

### PR DESCRIPTION
**Pull request by matthiasbeyer**
Actually, we're calling the command on an object. This object is the
shell_surface instance we want to alter here, not the first argument
passed to the command, right? So we should use this object to do our
transformation on, not something else.

@neithernut Please review.

### CODE REVIEW

**Comment by TheNeikos** [comment]
ACK

**Comment by manuelmessner** [comment]
ACK

**Comment by neithernut** [comment]
The fist two arguments _any_ command callback set on an object see's is the object itself, the second one is the method name. So this code will fail horribly for any case where the code is actually called.

**Comment by matthiasbeyer** [comment]
What "code" are you talking about here?

The first argument is the object, of course. That's the whole point of the PR, as we want to call an method on this object, we need to check it. Afterwards we check the arguments of the command. So I don't get where my error is...?

File: src/compositor/wayland/shell_surface.c [hunk 1]
```diff
@@ -422,19 +422,23 @@ static int
...
+        return -EINVAL;
+    }
+
+    struct wl_resource* r = (struct wl_resource*)
+                            ws_value_object_id_get(&stack[0].object_id);
+
     stack += 2; // Ignore the object and the command name
 
-    if (ws_value_get_type(&stack[0].value) != WS_VALUE_TYPE_OBJECT_ID ||
+    if (ws_value_get_type(&stack[0].value) != WS_VALUE_TYPE_INT ||
```
Comment (neithernut) [inline src/compositor/wayland/shell_surface.c:434]: stack[0] will _always_ be an object id!

File: src/compositor/wayland/shell_surface.c [hunk 2]
```diff
@@ -422,19 +422,23 @@ static int
...
             ws_value_get_type(&stack[1].value) != WS_VALUE_TYPE_INT ||
```
Comment (neithernut) [inline src/compositor/wayland/shell_surface.c:435]: stack[1] will _always_ be a string!

**Comment by neithernut** [comment]
Your entire argument list is offset!

File: src/compositor/wayland/shell_surface.c (hunk 1 above)
Comment (matthiasbeyer) [inline src/compositor/wayland/shell_surface.c:434]: Not after `stack += 2`.

File: src/compositor/wayland/shell_surface.c (hunk 2 above)
Comment (matthiasbeyer) [inline src/compositor/wayland/shell_surface.c:435]: Not after `stack += 2`.

**Comment by neithernut** [comment]
Ah, _now_ I get it. Ok, than I can ACK this as well.

**Pull request merged**